lms/services/blackboard_api/_schemas.py

```diff
@@ -8,21 +8,37 @@
 from lms.validation._base import RequestsResponseSchema
 
 
+class _FileSchema(Schema):
+    """Shared base schema for Blackboard file dicts."""
+
+    class Meta:
+        unknown = EXCLUDE
+
+    id = fields.Str(required=True)
+    display_name = fields.Str(data_key="name", required=True)
+    updated_at = fields.Str(data_key="modified", required=True)
+
+    @post_load
+    def post_load(self, data, **_kwargs):  # pylint:disable=no-self-use
+        data["id"] = f"blackboard://content-resource/{data['id']}/"
+        return data
+
+
 class BlackboardListFilesSchema(RequestsResponseSchema):
     """Schema for Blackboard API /courses/{courseId}/resources responses."""
 
-    class FileSchema(Schema):
-        """Schema for an individual file dict inside the "results" list."""
+    results = fields.List(fields.Nested(_FileSchema), required=True)
 
-        class Meta:
-            unknown = EXCLUDE
+    @post_load
+    def post_load(self, data, **_kwargs):  # pylint:disable=no-self-use
+        return data["results"]
 
-        id = fields.Str(required=True)
-        display_name = fields.Str(data_key="name", required=True)
-        updated_at = fields.Str(data_key="modified", required=True)
 
-    results = fields.List(fields.Nested(FileSchema), required=True)
+class BlackboardPublicURLSchema(RequestsResponseSchema, _FileSchema):
+    """Schema for Blackboard /courses/{courseId}/resources/{resourceId} responses."""
+
+    downloadUrl = fields.Str(required=True)
 
     @post_load
-    def make_object(self, data, **_kwargs):  # pylint:disable=no-self-use
-        return data["results"]
+    def post_load(self, data, **_kwargs):
+        return data["downloadUrl"]
```

Comment:
In line with the design of the existing `BlackboardListFilesSchema`, `BlackboardPublicURLSchema` does all the work and just returns the download URL string directly rather than returning a dict

Reply:
> In line with the design of the existing `BlackboardListFilesSchema`, `BlackboardPublicURLSchema` does all the work and just returns the download URL string directly rather than returning a dict

I think `all the work` here is accessing one field explicit declared in the schema. If we needed to do "blackboard course copy" kind of hack, adding `size` here would mean changing all the users of the schema or alternatively using a different schema per use case of the same API endpoint which I don't think it's very flexible design.

I reckon the schema should be more coupled to the endpoint it talks to rather than  to the use case we are giving that endpoint at the moment, I think that work for the service.

Reply:
I think it's potentially true that a schema that's closer to the Blackboard API endpoint is more reusable in cases where we have multiple uses for the same API endpoint or where we're doing some more complicated logic with the responses. But as the Blackboard API client stands right now though, splitting the work between the schema and the service would just complicate the code and tests (discussed in more detail here: <https://github.com/hypothesis/lms/pull/2800#discussion_r652525397>). I think maybe we can be led by the code and change this responsibility split between schema and service in future if it makes the code and tests better for future needs?